### PR TITLE
StateMinerInfo struct

### PR DESCRIPTION
**Pull request by turuslan**
json rpc requires specific struct, not `Universal`.
remove `seal_proof_type`.

### CODE REVIEW

File: core/primitives/tipset/tipset.cpp [hunk 1]
```diff
@@ -57,12 +57,10 @@ namespace fc::primitives::tipset {
   using vm::message::SignedMessage;
   using vm::message::UnsignedMessage;
 
-  MessageVisitor::~MessageVisitor() {
-    if (state_tree) {
-      delete state_tree;
-      state_tree = nullptr;
-    }
-  }
+  MessageVisitor::~MessageVisitor() = default;
```
Comment (wer1st) [inline core/primitives/tipset/tipset.cpp:60]: Could we move it to .hpp?

File: core/api/full_node/node_api.hpp [hunk 2]
```diff
@@ -265,6 +266,17 @@ namespace fc::api {
     TokenAmount max_fee;
   };
 
+  struct MinerInfo {
+    Address owner;
+    Address worker;
+    std::vector<Address> control;
+    Buffer peer_id;
+    std::vector<Multiaddress> multiaddrs;
+    RegisteredPoStProof window_post_proof_type{RegisteredPoStProof::kUndefined};
```
Comment (wer1st) [inline core/api/full_node/node_api.hpp:275]: window_post_proof_type is used starting v3. For v0 & v2 we use `getRegisteredWindowPoStProof(seal_proof_type)`, is it ok for you?

File: core/primitives/tipset/tipset.cpp (hunk 1 above)
Comment (turuslan) [reply core/primitives/tipset/tipset.cpp:60]: No, because of unique_ptr to forward-declared type

File: core/api/full_node/node_api.hpp (hunk 2 above)
Comment (turuslan) [reply core/api/full_node/node_api.hpp:275]: v0/v2 `MinerActorState::getInfo()` sets window_post_proof_type from seal_proof_type

Comment (wer1st) [reply core/api/full_node/node_api.hpp:275]: Yes, my question is: is it ok that getInfo() sets window_post_proof_type from seal_proof_type? Is it suitable for you in cases when api is called for v0 or v2?

Comment (turuslan) [reply core/api/full_node/node_api.hpp:275]: Yes, suitable.

**Review by wer1st — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**